hangman.py

```diff
@@ -0,0 +1,250 @@
+    my_word: string with _ characters, current guess of secret word;
+    returns: boolean - True, prints information message
+    and calls func get_guessed_word, if user_guess in word,
+    otherwise False, information message and take away attempts (guesses);
+    """
+    global letters_guessed, guesses
+
+    is_guessed = user_guess in my_word
+    if is_guessed:  # if user guessed a letter
+        print('Good guess:', get_guessed_word(my_word, letters_guessed))
+    else:  # if user didn't guess a letter
+        guesses -= 2 if user_guess in VOWELS else 1
+        if guesses >= 0:
+            print(f'Oops! That letter is not in my word.\n'
+                  f'Please guess a letter: {get_guessed_word(my_word, letters_guessed)}')
+    return is_guessed
+
+
+print('Welcome to the game Hangman!')
+print("I'm thinking of a word that is", len(word), "letters long.\nYou have ", warnings, "warnings and", guesses,
+      "guesses left")
+
+
+def hangman(secret_word, game_mode):
+    """
+    secret_word: string, the secret word to guess.
+    game_mode: 1 - without hints, 2 - with hints
+    returns:
+    Starts up an interactive game of Hangman.
+
+    """
+    global warnings, guesses, letters, letters_guessed
+    if 0 or guesses <= 0:  # if the user has run out of guesses
+        return print('-' * 20, f'\nSorry, you ran out of guesses/warnings. The word was else:\n'
+                               f'"{secret_word}"')
+
+    print('-' * 20, f'\nYou have {guesses} guesses left.\nAvailable letters: {letters}')
+    user_guess = str(input('Please guess a letter: ')).lower()
+
+    if user_guess == HINT and game_mode:  # if the user has input "*" symbol
+        random_words = show_possible_matches(get_guessed_word(secret_word, letters_guessed))
+        print('Possible word matches are:', ', '.join(random_words))  # show 10 possible matches
+        return hangman(secret_word, game_mode)
+
+    elif not get_warnings(user_guess, secret_word) or not get_guesses(user_guess, secret_word) \
+            or not is_word_guessed(secret_word, word_letters_guessed):
+        return hangman(secret_word, game_mode)
+
+
+def match_with_gaps(my_word, other_word):
+    """
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    """
+
+    counter = 0
+    my_word = my_word.replace(' ', '')
+    if len(my_word) == len(other_word):
+        for i in range(len(my_word)):
+            if my_word[i] == '_' or my_word[i] == other_word[i]:  # checks if letter in word fits letter in other word
+                continue
+            else:
+                counter += 1
+        if counter == 0:
+            return True
+        else:
+            return False
```

Comment:
There's no need in having that `counter`, just perform a `return False` in place where you increase it by 1